src/client/src/database.rs

```diff
@@ -148,42 +124,35 @@ impl TryFrom<api::v1::ObjectResult> for ObjectResult {
             .fail();
         }
 
-        let obj_result = object_result.result.context(error::MissingResultSnafu {
-            name: "result".to_string(),
-            expected: 1_usize,
-            actual: 0_usize,
-        })?;
-        Ok(match obj_result {
-            object_result::Result::Mutate(mutate) => ObjectResult::Mutate(mutate),
-            object_result::Result::FlightData(flight_data) => {
-                let flight_messages = raw_flight_data_to_message(flight_data.raw_data)
-                    .context(error::ConvertFlightDataSnafu)?;
-                ObjectResult::FlightData(flight_messages)
-            }
-        })
-    }
-}
-
-impl TryFrom<ObjectResult> for Output {
-    type Error = error::Error;
+        let flight_messages = raw_flight_data_to_message(object_result.flight_data)
+            .context(ConvertFlightDataSnafu)?;
 
-    fn try_from(value: ObjectResult) -> Result<Self> {
-        let output = match value {
-            ObjectResult::Mutate(mutate) => {
-                if mutate.failure != 0 {
-                    return error::MutateFailureSnafu {
-                        failure: mutate.failure,
-                    }
-                    .fail();
+        let output = if let Some(FlightMessage::AffectedRows(rows)) = flight_messages.get(0) {
```

Comment:
Is it possible for `flight_messages` to be empty?

Reply:
If `flight_messages` is empty, it will execute the "else" branch, and finally produce an empty `Recordbatches`.